core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -200,17 +176,50 @@ class CassandraJournal(cfg: Config) extends AsyncWriteJournal with CassandraReco
               case write :: remainder => writeMessages(write).flatMap(result => rec(remainder, result :: acc))
               case Nil                => Future.successful(acc.reverse)
             }
+
           rec(groups, Nil)
         }
 
-      result.onComplete { _ =>
+      val resultWithNotification = result.flatMap({ _ =>
+        val messagesByTag: Map[String, Seq[Serialized]] = serialized
+          .flatMap(_.payload)
+          .flatMap(s => s.tags.map((_, s)))
+          .groupBy(_._1).mapValues(_.map(_._2))
+
+        // todo make configurable and/or cirvuit break tag writes
+        implicit val timeout = Timeout(1.second)
+        val acks = messagesByTag.map {
+          case (tag, writes) =>
+            val tagWriter: ActorRef = tagActors.computeIfAbsent(tag, new JFunction[String, ActorRef] {
+              def apply(t: String): ActorRef = {
+                val tagSession = new TagWriterSession(
+                  tag,
+                  preparedWriteToTagView,
+                  session.executeWrite,
+                  session.selectResultSet,
+                  preparedWriteToTagProgress,
+                  preparedWriteToTagFirstBucket,
+                  preparedSelectTagProgress
+                )
+                context.actorOf(
+                  TagWriter.props(tagSession, tag, config.tagWriterSettings), s"tag-writer-$tag"
+                )
+              }
+            })
+            (tagWriter ? TagWrites(writes.toVector)).mapTo[TagWriteAck.type]
```

Comment:
why is the ack needed?

Reply:
I don't think it is, i can't remember why I added it, will remove